src/main/java/org/thingsboard/gateway/extensions/sigfox/conf/mapping/DataValueTransformer.java

```diff
@@ -0,0 +1,16 @@
+package org.thingsboard.gateway.extensions.sigfox.conf.mapping;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.PROPERTY,
+        property = "type")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = DoubleValueTransformer.class, name = DoubleValueTransformer.INT_TO_DOUBLE_TRANSFORMER_NAME)
+})
+public interface DataValueTransformer<T> {
+    T transform(T value);
```

Comment:
add 4 methods instead of this one:

T transform(String value);
T transform(Boolean value);
T transform(Double value);
T transform(Long value);

This will allow more flexible implementations of transformers in the future